lib/database/achievement.php

```diff
@@ -413,6 +413,12 @@ function AddEarnedAchievementJSON( $user, $achIDToAward, $isHardcore, $validatio
             $retVal[ 'Success' ] = FALSE;
             $retVal[ 'Error' ] = "User data cannot be found for $user";
         }
+        else if( $achData[ 'Flags' ] == 5 ) // Unofficial achievement
```

Comment:
Flags is a bitfield. You should be checking "($achData['Flags'] & 4) != 0". I'm not sure what the other bit is. The toolkit always sets it when committing achievements. (Core = 1|2, Unofficial = 1|4)